File: TechGig/Elections/code.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

void solve(int* row , int * col , int r , int c) {
    for ( int i = 0; i < r; i++) {
        sort(col, col + c,greater<int>());
        int temp = row[i];
        if ( temp > c ) {
            cout << "NO\n";
            return;
        }
        if ( temp != 0) {
            for(int j = 0; j < temp; j++) {
                if ( col[j] > r) {
                    cout << "NO\n";
                    return;
                }
                col[j]--;
            }
        }
    }
    for ( int i = 0; i < c; i++) {
        if ( col[i] != 0 ) {
            cout << "NO\n";
            return;
        }
    }
    cout << "YES\n";
    return;
}

int main() {
    ios_base::sync_with_stdio(false); 
    cin.tie(NULL); 
    int t, r, c;
    cin >> t;
    for ( int p = 0; p < t; p++) {
        cin >> r;
        cin >> c;
        int row[r] , col[c];
        for ( int i = 0; i < r; i++) {
            cin >> row[i];
        }
        for ( int i = 0; i < c; i++) {
            cin >> col[i];
        }
        sort(row, row + r);
        solve(row ,col , r ,c);
    }
    return 0;
    
}

File: TechGig/Elections/code.py
def solve(row,col):
    for i in range(0,len(row)):
        col.sort(reverse = True)
        temp = row[i]
        if temp > len(col):
            return False
        if temp != 0:
            for j in range(0,temp):
                col[j] = col[j] - 1

    for i in range(0,len(col)):
        if col[i] != 0:
            return False

    return True

for t in range(int(input())):
    first = input()
    second = input().split(' ')
    third = input().split(' ')
    row = []
    col = []
    print(type(second))
    for i in range(len(second)):
        row.append(int(second[i]))
    for i in range(len(third)):
        col.append(int(third[i]))
    
    if solve(row,col) == True:
        print("YES\n")
    else:
        print("NO\n")
